lib/compiler-llvm/src/abi/mod.rs

```diff
@@ -0,0 +1,85 @@
+// LLVM implements part of the ABI lowering internally, but also requires that
+// the user pack and unpack values themselves sometimes. This can help the LLVM
+// optimizer by exposing operations to the optimizer, but it requires that the
+// frontend know exactly what IR to produce in order to get the right ABI.
+
+#![deny(dead_code, missing_docs)]
+
+use crate::translator::intrinsics::Intrinsics;
+use inkwell::{
+    attributes::{Attribute, AttributeLoc},
+    builder::Builder,
+    context::Context,
+    targets::TargetMachine,
+    types::FunctionType,
+    values::{BasicValueEnum, CallSiteValue, FunctionValue, PointerValue},
+};
+use wasmer_compiler::CompileError;
+use wasmer_types::FunctionType as FuncSig;
+
+mod aarch64_systemv;
+mod x86_64_systemv;
+
+use aarch64_systemv::Aarch64SystemV;
+use x86_64_systemv::X86_64SystemV;
+
+pub fn get_abi(target_machine: &TargetMachine) -> Box<dyn Abi> {
+    if target_machine
+        .get_triple()
+        .as_str()
+        .to_string_lossy()
+        .starts_with("aarch64")
```

Comment:
Why not `get_cpu`?

https://thedan64.github.io/inkwell/inkwell/targets/struct.TargetMachine.html#method.get_cpu

Reply:
I tried that! It returns the string `"generic"`.